--- events-manager/admin/em-bookings.php.old	2016-04-14 22:14:46.000000000 +0300
+++ events-manager/admin/em-bookings.php	2017-04-27 17:56:02.408921800 +0300
@@ -85,7 +85,7 @@
 	?>
 	<div class='wrap'>
 		<?php if( is_admin() ): ?><h1><?php else: ?><h2><?php endif; ?>		
-  			<?php echo sprintf(__('Manage %s Bookings', 'events-manager'), "'{$EM_Event->event_name}'"); ?>
+  			<?php echo apply_filters('translate_text', sprintf('Manage %s Bookings', "'{$EM_Event->event_name}'"), 'ru'); ?>
   			<a href="<?php echo $EM_Event->get_permalink(); ?>" class="<?php echo $header_button_classes; ?>"><?php echo sprintf(__('View %s','events-manager'), __('Event', 'events-manager')) ?></a>
   			<a href="<?php echo $EM_Event->get_edit_url(); ?>" class="<?php echo $header_button_classes; ?>"><?php echo sprintf(__('Edit %s','events-manager'), __('Event', 'events-manager')) ?></a>
   			<?php if( locate_template('plugins/events-manager/templates/csv-event-bookings.php', false) ): //support for legacy template ?>
@@ -95,7 +95,7 @@
 		<?php if( !is_admin() ): ?></h2><?php else: ?></h1><?php endif; ?>
   		<?php if( !is_admin() ) echo $EM_Notices; ?>  
 		<div>
-			<p><strong><?php esc_html_e('Event Name','events-manager'); ?></strong> : <?php echo esc_html($EM_Event->event_name); ?></p>
+			<p><strong><?php esc_html_e('Event Name','events-manager'); ?></strong> : <?php echo esc_html(apply_filters('translate_text', $EM_Event->event_name, 'ru')); ?></p>
 			<p>
 				<strong><?php esc_html_e('Availability','events-manager'); ?></strong> : 
 				<?php echo $EM_Event->get_bookings()->get_booked_spaces() . '/'. $EM_Event->get_spaces() ." ". __('Spaces confirmed','events-manager'); ?>
@@ -142,7 +142,7 @@
 	?>
 	<div class='wrap'>
 		<?php if( is_admin() ): ?><h1><?php else: ?><h2><?php endif; ?>
-  			<?php echo sprintf(__('Ticket for %s', 'events-manager'), "'{$EM_Event->name}'"); ?>
+  			<?php echo apply_filters('translate_text', sprintf('Ticket for %s', "'{$EM_Event->name}'"), 'ru'); ?>
   			<a href="<?php echo $EM_Event->get_edit_url(); ?>" class="<?php echo $header_button_classes; ?>"><?php esc_html_e('View/Edit Event','events-manager') ?></a>
   			<a href="<?php echo $EM_Event->get_bookings_url(); ?>" class="<?php echo $header_button_classes; ?>"><?php esc_html_e('View Event Bookings','events-manager') ?></a>
   		
@@ -150,7 +150,7 @@
   		<?php if( !is_admin() ) echo $EM_Notices; ?>
 		<div>
 			<table>
-				<tr><td><?php echo __('Name','events-manager'); ?></td><td></td><td><?php echo $EM_Ticket->ticket_name; ?></td></tr>
+				<tr><td><?php echo __('Name','events-manager'); ?></td><td></td><td><?php echo apply_filters('translate_text', $EM_Ticket->ticket_name, 'ru'); ?></td></tr>
 				<tr><td><?php echo __('Description','events-manager'); ?>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td></td><td><?php echo ($EM_Ticket->ticket_description) ? $EM_Ticket->ticket_description : '-'; ?></td></tr>
 				<tr><td><?php echo __('Price','events-manager'); ?></td><td></td><td><?php echo ($EM_Ticket->ticket_price) ? $EM_Ticket->ticket_price : '-'; ?></td></tr>
 				<tr><td><?php echo __('Spaces','events-manager'); ?></td><td></td><td><?php echo ($EM_Ticket->ticket_spaces) ? $EM_Ticket->ticket_spaces : '-'; ?></td></tr>
@@ -204,7 +204,7 @@
 							$localised_end_date = date_i18n(get_option('dbem_date_format'), $EM_Event->end);
 							?>
 							<table>
-								<tr><td><strong><?php esc_html_e('Name','events-manager'); ?></strong></td><td><a class="row-title" href="<?php echo $EM_Event->get_bookings_url(); ?>"><?php echo ($EM_Event->event_name); ?></a></td></tr>
+								<tr><td><strong><?php esc_html_e('Name','events-manager'); ?></strong></td><td><a class="row-title" href="<?php echo $EM_Event->get_bookings_url(); ?>"><?php echo apply_filters('translate_text', $EM_Event->event_name, 'ru'); ?></a></td></tr>
 								<tr>
 									<td><strong><?php esc_html_e('Date/Time','events-manager'); ?>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</strong></td>
 									<td>
@@ -309,7 +309,7 @@
 									<tbody>
 										<?php foreach($EM_Booking->get_tickets_bookings()->tickets_bookings as $EM_Ticket_Booking): /* @var $EM_Ticket_Booking EM_Ticket_Booking */ ?>
 										<tr>
-											<td class="ticket-type"><a class="row-title" href="<?php echo em_add_get_params($EM_Event->get_bookings_url(), array('ticket_id'=>$EM_Ticket_Booking->ticket_id)); ?>"><?php echo $EM_Ticket_Booking->get_ticket()->ticket_name ?></a></td>
+											<td class="ticket-type"><a class="row-title" href="<?php echo em_add_get_params($EM_Event->get_bookings_url(), array('ticket_id'=>$EM_Ticket_Booking->ticket_id)); ?>"><?php echo apply_filters('translate_text', $EM_Ticket_Booking->get_ticket()->ticket_name, 'ru'); ?></a></td>
 											<td>
 												<span class="em-booking-single-info"><?php echo $EM_Ticket_Booking->get_spaces(); ?></span>
 												<div class="em-booking-single-edit"><input name="em_tickets[<?php echo $EM_Ticket_Booking->ticket_id; ?>][spaces]" class="em-ticket-select" id="em-ticket-spaces-<?php echo $EM_Ticket_Booking->ticket_id; ?>" value="<?php echo $EM_Ticket_Booking->get_spaces(); ?>" /></div>
@@ -325,7 +325,7 @@
 											<?php foreach($EM_Event->get_bookings()->get_tickets()->tickets as $EM_Ticket): /* @var $EM_Ticket EM_Ticket */ ?>
 												<?php if( !in_array($EM_Ticket->ticket_id, $shown_tickets) ): ?>
 												<tr>
-													<td class="ticket-type"><a class="row-title" href="<?php echo em_add_get_params($EM_Event->get_bookings_url(), array('ticket_id'=>$EM_Ticket->ticket_id)); ?>"><?php echo $EM_Ticket->ticket_name ?></a></td>
+													<td class="ticket-type"><a class="row-title" href="<?php echo em_add_get_params($EM_Event->get_bookings_url(), array('ticket_id'=>$EM_Ticket->ticket_id)); ?>"><?php echo apply_filters('translate_text', $EM_Ticket->ticket_name, 'ru'); ?></a></td>
 													<td>
 														<span class="em-booking-single-info">0</span>
 														<div class="em-booking-single-edit"><input name="em_tickets[<?php echo $EM_Ticket->ticket_id; ?>][spaces]" class="em-ticket-select" id="em-ticket-spaces-<?php echo $EM_Ticket_Booking->ticket_id; ?>" value="0" /></div>
--- events-manager/admin/bookings/em-events.php.old	2015-12-18 14:27:56.000000000 +0200
+++ events-manager/admin/bookings/em-events.php	2017-04-27 17:56:02.763957300 +0300
@@ -115,7 +115,7 @@
 						<tr <?php echo "$class $style"; ?>>
 							<td>
 								<strong>
-									<?php echo $event->output('#_BOOKINGSLINK'); ?>
+									<?php echo apply_filters('translate_text', $event->output('#_BOOKINGSLINK'), 'ru'); ?>
 								</strong>
 								&ndash; 
 								<?php esc_html_e("Booked Spaces",'events-manager') ?>: <?php echo $event->get_bookings()->get_booked_spaces()."/".$event->get_spaces() ?>
--- events-manager/classes/em-bookings-table.php.old	2015-12-18 14:27:56.000000000 +0200
+++ events-manager/classes/em-bookings-table.php	2017-04-27 17:56:04.018082700 +0300
@@ -66,7 +66,7 @@
 		//Set basic vars
 		$this->order = ( !empty($_REQUEST ['order']) && $_REQUEST ['order'] == 'DESC' ) ? 'DESC':'ASC';
 		$this->orderby = ( !empty($_REQUEST ['orderby']) ) ? sanitize_sql_orderby($_REQUEST['orderby']):'booking_name';
-		$this->limit = ( !empty($_REQUEST['limit']) && is_numeric($_REQUEST['limit'])) ? $_REQUEST['limit'] : 20;//Default limit
+		$this->limit = ( !empty($_REQUEST['limit']) && is_numeric($_REQUEST['limit'])) ? $_REQUEST['limit'] : 100;//Default limit
 		$this->page = ( !empty($_REQUEST['pno']) && is_numeric($_REQUEST['pno']) ) ? $_REQUEST['pno']:1;
 		$this->offset = ( $this->page > 1 ) ? ($this->page-1)*$this->limit : 0;
 		$this->scope = ( !empty($_REQUEST['scope']) && array_key_exists($_REQUEST ['scope'], em_get_scopes()) ) ? sanitize_text_field($_REQUEST['scope']):get_option('dbem_default_bookings_search','future');
@@ -408,7 +408,7 @@
 						<input name="pno" type="hidden" value="1" />
 						<input id="post-query-submit" class="button-secondary" type="submit" value="<?php esc_attr_e( 'Filter' )?>" />
 						<?php if( $EM_Event !== false ): ?>
-						<?php esc_html_e('Displaying Event','events-manager'); ?> : <?php echo esc_html($EM_Event->event_name); ?>
+						<?php esc_html_e('Displaying Event','events-manager'); ?> : <?php echo esc_html(apply_filters('translate_text', $EM_Event->event_name, 'ru')); ?>
 						<?php elseif( $EM_Person !== false ): ?>
 						<?php esc_html_e('Displaying User','events-manager'); echo ' : '.esc_html($EM_Person->get_name()); ?>
 						<?php endif; ?>
@@ -540,9 +540,9 @@
 				$val = esc_html($EM_Booking->get_person()->last_name);
 			}elseif($col == 'event_name'){
 				if( $csv ){
-					$val = $EM_Booking->get_event()->event_name;
+					$val = apply_filters('translate_text', $EM_Booking->get_event()->event_name, 'ru');
 				}else{
-					$val = '<a href="'.$EM_Booking->get_event()->get_bookings_url().'">'. esc_html($EM_Booking->get_event()->event_name) .'</a>';
+					$val = '<a href="'.$EM_Booking->get_event()->get_bookings_url().'">'. esc_html(apply_filters('translate_text', $EM_Booking->get_event()->event_name, 'ru')) .'</a>';
 				}
 			}elseif($col == 'event_date'){
 				$val = $EM_Booking->get_event()->output('#_EVENTDATES');
--- events-manager/em-actions.php.old	2017-04-30 09:05:27.517204300 +0300
+++ events-manager/em-actions.php	2017-04-30 09:05:59.030006700 +0300
@@ -250,7 +250,7 @@
 		if ( $_REQUEST['action'] == 'booking_add') {
 			//ADD/EDIT Booking
 			ob_start();
-			if( !defined('WP_CACHE') || !WP_CACHE ) em_verify_nonce('booking_add');
+			// if( !defined('WP_CACHE') || !WP_CACHE ) em_verify_nonce('booking_add');
 			if( !is_user_logged_in() || get_option('dbem_bookings_double') || !$EM_Event->get_bookings()->has_booking(get_current_user_id()) ){
 			    $EM_Booking->get_post();
 				$post_validation = $EM_Booking->validate();
--- em-booking.php.old  2018-07-26 09:11:19.820381400 +0300
+++ em-booking.php      2018-07-26 09:04:22.319635100 +0300
@@ -947,9 +947,9 @@
         * @param string $note
         * @return mixed
         */
-       function add_note( $note_text ){
+       function add_note( $note_text, $force = false ){
                global $wpdb;
-               if( $this->can_manage() ){
+               if( $force || $this->can_manage() ){
                        $this->get_notes();
                        $note = array('author'=>get_current_user_id(),'note'=>$note_text,'timestamp'=>current_time('timestamp'));
                        $this->notes[] = wp_kses_data($note);
