--- events-manager/classes/em-bookings-table.php.old	2015-12-18 14:27:56.000000000 +0200
+++ events-manager/classes/em-bookings-table.php	2016-07-17 09:19:32.558516200 +0300
@@ -66,7 +66,7 @@
 		//Set basic vars
 		$this->order = ( !empty($_REQUEST ['order']) && $_REQUEST ['order'] == 'DESC' ) ? 'DESC':'ASC';
 		$this->orderby = ( !empty($_REQUEST ['orderby']) ) ? sanitize_sql_orderby($_REQUEST['orderby']):'booking_name';
-		$this->limit = ( !empty($_REQUEST['limit']) && is_numeric($_REQUEST['limit'])) ? $_REQUEST['limit'] : 20;//Default limit
+		$this->limit = ( !empty($_REQUEST['limit']) && is_numeric($_REQUEST['limit'])) ? $_REQUEST['limit'] : 100;//Default limit
 		$this->page = ( !empty($_REQUEST['pno']) && is_numeric($_REQUEST['pno']) ) ? $_REQUEST['pno']:1;
 		$this->offset = ( $this->page > 1 ) ? ($this->page-1)*$this->limit : 0;
 		$this->scope = ( !empty($_REQUEST['scope']) && array_key_exists($_REQUEST ['scope'], em_get_scopes()) ) ? sanitize_text_field($_REQUEST['scope']):get_option('dbem_default_bookings_search','future');
@@ -393,7 +393,7 @@
 							<option value="10">10</option>
 							<option value="25">25</option>
 							<option value="50">50</option>
-							<option value="100">100</option>
+							<option value="200">200</option>
 						</select>
 						<select name="status">
 							<?php
--- events-manager/classes/em-tickets-bookings.php.old	2016-04-14 22:14:46.000000000 +0300
+++ events-manager/classes/em-tickets-bookings.php	2016-07-20 15:42:24.557778200 +0300
@@ -34,7 +34,7 @@
 		if($object){
 			if( is_object($object) && get_class($object) == "EM_Booking"){
 				$this->booking = $object;
-				$sql = "SELECT * FROM ". EM_TICKETS_BOOKINGS_TABLE ." WHERE booking_id ='{$this->booking->booking_id}'";
+				$sql = "SELECT * FROM ". EM_TICKETS_BOOKINGS_TABLE ." WHERE booking_id ='{$this->booking->booking_id}' ORDER BY ticket_id ASC";
 			}elseif( is_object($object) && get_class($object) == "EM_Ticket"){
 				$this->ticket = $object;
 				$sql = "SELECT * FROM ". EM_TICKETS_BOOKINGS_TABLE ." WHERE ticket_id ='{$this->ticket->ticket_id}'";
